packages/cli/src/commands/console.ts

```diff
@@ -65,11 +66,44 @@ export class Console extends Command {
       .on('exit', () => watchers.forEach((watcher) => watcher.close()))
   }
 
+  private static setupSelfRolledHistory(repl: any, path: string) {
+    function init() {
+      try {
+        const history = fs.readFileSync(path, {encoding: 'utf8'})
+        const nonEmptyLines = history.split(os.EOL).filter((line) => line.trim())
+        repl.history.push(...nonEmptyLines.reverse())
+      } catch (err) {
+        if (err.code !== 'ENOENT') {
+          throw err
+        }
+      }
+    }
+
+    function onExit() {
+      const addedHistory = repl.lines.join(os.EOL)
+      fs.appendFileSync(path, addedHistory)
+    }
+
+    init()
+    repl.on('exit', onExit)
+  }
```

Comment:
This looks nice! Are there any known limitations or gotcha's with this vs the built in `setupHistory()` in Node 12+?

Reply:
Thx 😄 Compared to v12's `setupHistory`, history is saved in reverse order. This will break when a user switches between v10 and v12. If we were to fix this, we'd need to prepend the file instead of append, making this rather costly for large histories.

Other than that, it basically felt like native history - although there must be some differences, as Node v10's internal implementation was quite a big larger.